packages/orama/src/types.ts

```diff
@@ -438,19 +495,26 @@ export type Components = Partial<ObjectComponents & FunctionComponents & SingleO
 export const kInsertions = Symbol('orama.insertions')
 export const kRemovals = Symbol('orama.removals')
 
-type ProvidedTypes = Partial<{ Schema: Schema; Index: OpaqueIndex; DocumentStore: OpaqueDocumentStore }>
+type ProvidedTypes = Partial<{
+  Schema: Schema;
+  Index: OpaqueIndex;
+  DocumentStore: OpaqueDocumentStore;
+  Sorter: OpaqueSorter;
+}>
 
-interface Data<I extends OpaqueIndex, D extends OpaqueDocumentStore> {
+interface Data<I extends OpaqueIndex, D extends OpaqueDocumentStore, S extends OpaqueSorter> {
   index: I
   docs: D
+  sorter: S
```

Comment:
Here the name does not follow the component it belongs to. For readability I would rename this to `sorting`.